redfish-core/include/utils/hw_isolation.hpp

```diff
@@ -334,5 +336,310 @@ inline void processHardwareIsolationReq(
+                        {
+                            BMCWEB_LOG_ERROR(
+                                "Failed to get the Timestamp from object: {}",
+                                hwStatusEventObj);
+                            messages::internalError(asyncResp->res);
+                            return;
+                        }
+                        condition["Timestamp"] =
+                            redfish::time_utils::getDateTimeStdtime(
+                                static_cast<std::time_t>(*timestamp));
+                    }
+                    else if (property.first == "Message")
+                    {
+                        const std::string* msgPropVal =
+                            std::get_if<std::string>(&property.second);
+                        if (msgPropVal == nullptr)
+                        {
+                            BMCWEB_LOG_ERROR(
+                                "Failed to get the Message from object: {}",
+                                hwStatusEventObj);
+                            messages::internalError(asyncResp->res);
+                            return;
+                        }
+
+                        const redfish::registries::Message* msgReg =
+                            registries::getMessage(
+                                "OpenBMC.0.2.HardwareIsolationReason");
+
+                        if (msgReg == nullptr)
+                        {
+                            BMCWEB_LOG_ERROR(
+                                "Failed to get the HardwareIsolationReason message registry to add in the condition");
+                            messages::internalError(asyncResp->res);
+                            return;
+                        }
+
+                        // Prepare MessageArgs as per defined in the
+                        // MessageRegistries
+                        std::vector<std::string> messageArgs{*msgPropVal};
+
+                        // Fill the "msgPropVal" as reason
+                        std::string message = msgReg->message;
+                        int i = 0;
+                        for (const std::string& messageArg : messageArgs)
+                        {
+                            std::string argIndex = "%" + std::to_string(++i);
+                            size_t argPos = message.find(argIndex);
+                            if (argPos != std::string::npos)
+                            {
+                                message.replace(argPos, argIndex.length(),
+                                                messageArg);
+                            }
+                        }
+                        // Severity will be added based on the event
+                        // object property
+                        condition["Message"] = message;
+                        condition["MessageArgs"] = messageArgs;
+                        condition["MessageId"] =
+                            "OpenBMC.0.2.HardwareIsolationReason";
```

Comment:
Shouldn't this match the message registry version that added this message? (i.e. OpenBMC.0.5.0)

Reply:
I am not sure. @baemyung Does it need to be updated to OpenBMC.0.5.0

Reply:
Since the current version 0.5.0, it may be good to use it.
However, the code `getMessage()` seems not utilizing the version.  So, either seems fine.

https://github.com/ibm-openbmc/bmcweb/blob/a33fa9a4d1528126ba92b9290915811ef1132201/redfish-core/lib/log_services.hpp#L124

Reply:
Will keep it the same for now. By the time I address the other comments, I will include this too.